pyscriptjs/src/main.ts

```diff
@@ -359,10 +415,15 @@ modules must contain a "plugin" attribute. For more information check the plugin
     }
 
     // lifecycle (7)
-    executeScripts(interpreter: InterpreterClient) {
+    async executeScripts(interpreter: InterpreterClient) {
         // make_PyScript takes an interpreter and a PyScriptApp as arguments
         this.PyScript = make_PyScript(interpreter, this);
         customElements.define('py-script', this.PyScript);
+        this.incrementNumPendingTags();
+        if (this.numPendingTags > 0) {
```

Comment:
in which circumstance this would ever be `false` as you just incremented that in the line before?

Reply:
It's just a sanity check wherever `decrementNumPendingTags` is called, so that we don't decrement when it is `0`.
Maybe I can encapsulate this check inside the function i.e. inside `decrementNumPendingTags` itself, what say?

Reply:
I've added another comment ... here my point is that you are checking an implementation detail (internal) right after knowing that `this.incrementNumPendingTags();` will add `+1` there ... let me write a basic example for you to explain why I think there's something wrong in here ...

```js
const controller = {
  i: 0,
  increment() { this.i++; },
  decrement() { this.i--; }
};

// do you think this code has good logic?
controller.increment();
if (controller.i > 0)
  controller.decrement();
```

Like I've said, this dance smells a bit to me ... nothing should check what the controller (main.ts) does internally but here in particular it looks like these 3 lines should just increment/decrement instead and let the controller do the dance (so I agree that checking that `numPendingTags` outside those 2 methods makes very little sense).

I'd probably move the check within the method that happens only when `numPendingTags === 0` too, still keeping the logic to decrement aside so that here you don't need to increment/decrement right away, you just call that method and if it's the only script let it bootstrap (this should work unless I'm missing something).

Reply:
I think it's fine to have the check but as I said above it should be an assert (i.e., interrupt the execution immediately), not a check which ignores the error, if it's there.

That said, I plan to refactor this logic in a subsequent PR.

Reply:
I encapsulated the check inside `decrementPendingTags`, do you want me to change it to an assert @antocuni?
The commit I am talking about is here: https://github.com/pyscript/pyscript/pull/1258/commits/fcee8facf4ec1a54d5af844e0a244afa97358b7a